openquake/hazardlib/geo/surface/gridded.py

```diff
@@ -125,7 +125,10 @@ def get_top_edge_depth(self):
             Float value, the vertical distance between the earth surface
             and the shallowest point in surface's top edge in km.
         """
-        raise NotImplementedError('GriddedSurface')
+        depths = []
+        for point in self.mesh:
+            depths.append(point.depth)
+        return min(depths)
```

Comment:
This is very inefficient. The right way is to return `self.mesh.depths.min()`

Reply:
Thanks Michele, I fixed this